### PR TITLE
redocly-cli: 1.6.0 -> 1.12.0, rename to redocly

### PR DESCRIPTION
**Pull request by szlend**
## Description of changes

- upgrade redocly-cli to v1.12.0 ([diff](https://github.com/Redocly/redocly-cli/compare/%40redocly/cli%401.6.0...%40redocly/openapi-core%401.12.0))
- rename redocly-cli -> redocly

Reasons for renaming:
- it's called redocly in other repos (see repology.org)
  - the naming difference is causing `r-ryantm` to miss updates
- the cli tool upstream is called `redocly` (see [README](https://github.com/Redocly/redocly-cli?tab=readme-ov-file#generate-api-reference-documentation))

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/re/redocly/package.nix [hunk 1]
```diff
@@ -49,7 +52,7 @@ buildNpmPackage rec {
     description = "Redocly CLI makes OpenAPI easy. Lint/validate to any standard, generate beautiful docs, and more.";
```
Comment (AndersonTorres) [inline pkgs/by-name/re/redocly/package.nix:52]: Remove the trailing dot

Comment (szlend) [reply pkgs/by-name/re/redocly/package.nix]: Done

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/by-name/re/redocly/package.nix [hunk 2]
```diff
@@ -32,24 +32,27 @@ buildNpmPackage rec {
...
-    $out/bin/redocly-cli --version
+    $out/bin/redocly --version
     runHook postInstallCheck
   '';
 
   doInstallCheck = true;
 
   meta = {
-    description = "Redocly CLI makes OpenAPI easy. Lint/validate to any standard, generate beautiful docs, and more.";
+    description = "Redocly CLI makes OpenAPI easy. Lint/validate to any standard, generate beautiful docs, and more";
```
Comment (Aleksanaa) [inline pkgs/by-name/re/redocly/package.nix:52]: Don't repeat package name.

```suggestion
    description = "A tool that makes OpenAPI easy. Lint/validate to any standard, generate beautiful docs, and more";
```

File: pkgs/by-name/re/redocly/package.nix [hunk 3]
```diff
@@ -32,24 +32,27 @@ buildNpmPackage rec {
...
-      $out/bin/redocly-cli \
+      $out/bin/redocly \
       --set-default REDOCLY_TELEMETRY off \
       --set-default CI true # Silence update messages
+
+    # Symlink for backwards compatibility. Remove after deprecation.
+    ln -s $out/bin/redocly $out/bin/redocly-cli
   '';
 
   installCheckPhase = ''
```
Comment (Aleksanaa) [inline pkgs/by-name/re/redocly/package.nix:43]: Would be better to migrate to `passthru.tests` with `testers.testVersion`

Comment (szlend) [reply pkgs/by-name/re/redocly/package.nix]: TIL, thanks!

File: nixos/doc/manual/release-notes/rl-2405.section.md [hunk 4]
```diff
@@ -456,6 +456,8 @@ The pre-existing [services.ankisyncd](#opt-services.ankisyncd.enable) has been m
 
 - The initrd ssh daemon module got a new option to add authorized keys via a list of files using `boot.initrd.network.ssh.authorizedKeyFiles`.
 
+- The `redocly-cli` package has been renamed to `redocly`. The old package was aliased for backwards compatibility.
+
```
Comment (SuperSandro2000) [inline nixos/doc/manual/release-notes/rl-2405.section.md:460]:
```suggestion
```
if we add an entry to aliases.nix, there is nothing people need to know about and they can fix the warning at some point.

File: pkgs/by-name/re/redocly/package.nix [hunk 5]
```diff
@@ -32,24 +34,25 @@ buildNpmPackage rec {
...
+      $out/bin/redocly \
       --set-default REDOCLY_TELEMETRY off \
       --set-default CI true # Silence update messages
-  '';
 
-  installCheckPhase = ''
-    runHook preInstallCheck
-    $out/bin/redocly-cli --version
-    runHook postInstallCheck
+    # Symlink for backwards compatibility. Remove after deprecation.
```
Comment (SuperSandro2000) [inline pkgs/by-name/re/redocly/package.nix:41]:
```suggestion
    # Symlink for backwards compatibility. Remove after NixOS 23.11
```

Comment (szlend) [reply pkgs/by-name/re/redocly/package.nix]: This is going into unstable/24.05. I'll remove it in 24.11.

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**